src/wallet/test/fuzz/rpc.cpp

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) 2021-2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <test/fuzz/rpc.h>
+
+enum class ChainType;
+
+namespace {
+struct RPCWalletFuzzTestingSetup : public TestingSetup {
+    RPCWalletFuzzTestingSetup(const ChainType chain_type, TestOpts opts) : TestingSetup{chain_type, opts}
+    {
+    }
+
+    void CallRPC(const std::string& rpc_method, const std::vector<std::string>& arguments)
+    {
+        JSONRPCRequest request;
+        request.context = &m_node;
+        request.strMethod = rpc_method;
+        try {
+            request.params = RPCConvertValues(rpc_method, arguments);
+        } catch (const std::runtime_error&) {
+            return;
+        }
+
+        tableRPC.execute(request);
+    }
+
+    std::vector<std::string> GetWalletRPCCommands() const
+    {
+        Span<const CRPCCommand> tableWalletRPC = wallet::GetWalletRPCCommands();
+        std::vector<std::string> supported_rpc_commands;
+        for (const auto& c : tableWalletRPC) {
+            tableRPC.appendCommand(c.name, &c);
+            supported_rpc_commands.push_back(c.name);
+        }
+        return supported_rpc_commands;
+    }
+};
+
+RPCWalletFuzzTestingSetup* rpc_wallet_testing_setup = nullptr;
+std::string g_limit_to_rpc_command_wallet;
+
+// RPC commands which are safe for fuzzing.
+const std::vector<std::string> WALLET_RPC_COMMANDS_NOT_SAFE_FOR_FUZZING{
+    "importwallet",
+    "loadwallet",
+};
```

Comment:
There are many more wallet rpcs that go to disk (e.g. `createwallet`). I would assume a lot of the rpcs that modify a wallet also flush the changes to disk. And I don't think it is possible to even have a wallet that doesn't exist on disk, so trying to avoid disk access can't work.

To work around this I would suggest to create (and delete at the end) a new wallet (e.g. `/tmp/fuzz-wallet-xyz`) each iteration and call wallet rpcs on that specific wallet (could be more than one).